arch/arm/src/stm32h7/stm32h7x3xx_rcc.c

```diff
@@ -886,6 +886,20 @@ void stm32_stdclockconfig(void)
           STM32_PWR_CR3_SMPSEXTHP | STM32_PWR_CR3_SMPSLEVEL_MASK);
       regval |= STM32_PWR_CR3_SCUEN;
       putreg32(regval, STM32_PWR_CR3);
+#elif defined(CONFIG_STM32H7_PWR_LDO_SUPPLY)
+      regval = getreg32(STM32_PWR_CR3);
+      regval &= ~(STM32_PWR_CR3_BYPASS | STM32_PWR_CR3_SCUEN |
+          STM32_PWR_CR3_SMPSEXTHP | STM32_PWR_CR3_SMPSLEVEL_MASK);
+      regval |= STM32_PWR_CR3_LDOEN;
+      putreg32(regval, STM32_PWR_CR3);
+#elif defined(CONFIG_STM32H7_PWR_SMPS_LDO) || \
+      defined(CONFIG_STM32H7_PWR_EXT_SMPS_LDO) || \
+      defined(CONFIG_STM32H7_PWR_EXT_SMPS_BYPASS) || \
+      defined(CONFIG_STM32H7_PWR_BYPASS)
+      /* These 4 cases of power supply mode have no macros created.
+       * Selecting one of these modes will cause a compile error.
+       */
+      #error Macro of this power supply mode is not created yet.
```

Comment:
What would a user be expected to do if they get this message?

Reply:
Since I cannot verify the operation of these modes, I hope that those who use them will add macros.

Reply:
The correct thing to do is: The Kconfig should name the 6 configurations available based on the package and IC Family  and the code should set them. The default should be what it was before this PR. 

Putting a partial solution and jeopardizing breaking existing boards is totally un-acceptable.

Reply:
@davids5
Before this PR, we could only use DIRECT_SMPS_SYPPLY mode or else.
And we just want to use LDO_SUPPLY mode.
Should we remove the 4 modes that no one is currently using?

And we are trying to be careful not to break the existing board.

Reply:
@jfbblue0922 I will spend some time in the next couple of days reviewing the data sheets the  and push some changes to this PR.